### DISCUSSION TITLE
Updated feed items, their hashes and extensions

### DISCUSSION THREAD

**Discussion by kour1er** [FreshRSS/FreshRSS#5602] [Q&A] [answered]
If my understanding is correct: when a feed item is updated, that is to say a previously posted item receives new content, an updated hash is created in the database --- reflecting a difference in the feed item.

In the context of an extension, this updated hash item will appear again when using entry_before_insert as the item is, in effect new. Would it be possible to provide to the extension an indication that this is an old updated item, rather than an entirely new one?

**Comment by Alkarex** [comment, marked as answer]
Currently, we emit the `entry_before_insert` event both for new articles (`INSERT`) and modified articles (`UPDATE`). We should differentiate the two cases.

Update: on the `edge` rolling release or from FreshRSS 1.24.0, one can test `$entry->isUpdated()`

Before FreshRSS 1.24.0, there is an undocumented trick:
```php
if (entry->id() == null) {
	//updated article
} else {
	//new article
}
```

**Reply by Alkarex** [reply]
Related to https://github.com/FreshRSS/FreshRSS/issues/1706#issuecomment-348772243

```php
$originalHash = $entry->hash();
// your code
$entry->_hash($originalHash);
```

Example https://github.com/kevinpapst/freshrss-dilbert/pull/1

**Reply by pe1uca** [reply]
If there's no current plan to implement a property `isNew`, can at least this hack be documented?  

I just went through the code to come with this workaround, and just when I was going to create a new issue found this discussion. 
This was my implementation since this is exactly how the feed controller makes the check.  
```php
$entryDAO = FreshRSS_Factory::createEntryDao();
$dbHashes = $entryDAO->listHashForFeedGuids($feed->id(), [$entry->guid()]);
// If the hash was loaded from the DB the entry already exists, so we don't need to process it again.
if (isset($dbHashes[$entry->guid()])) return $entry;
```

https://github.com/FreshRSS/FreshRSS/blob/86631d24b9b41e35cd915d253a2040bdb8dbbfba/app/Controllers/feedController.php#L571


There's this a few lines below that.  
Maybe something similar can be done to know if the entry is being updated: `$entry->_isNew(false);`
https://github.com/FreshRSS/FreshRSS/blob/86631d24b9b41e35cd915d253a2040bdb8dbbfba/app/Controllers/feedController.php#L577

**Reply by Alkarex** [reply]
Update: on the `edge` rolling release or from FreshRSS 1.24.0, one can test `$entry->isUpdated()`

**Comment by math-GH** [comment]
Is here anything left todo?

**Comment by kour1er** [comment]
The undocumented trick is working great for me - cheers
